erpnext/vehicles/doctype/vehicle_gate_pass/vehicle_gate_pass.js

```diff
@@ -24,14 +24,94 @@ erpnext.vehicles.VehicleGatePass = class VehicleGatePass extends erpnext.vehicle
 		});
 
 		me.frm.set_query("sales_invoice", function() {
-			var filters = {"docstatus": ['<', 2]};
-
-			filters['project'] = me.frm.doc.project;
+			return {
+				filters: {
+					project: me.frm.doc.project,
+					docstatus:  ['<', 2]
+				}
+			};
+		});
 
+		me.frm.set_query("vehicle_delivery", function() {
 			return {
-				filters: filters
+				filters: {
+					item_code : me.frm.doc.item_code,
+					docstatus: ['=', 1]
+				}
 			};
 		});
+
+		me.frm.set_query("vehicle_booking_order", function() {
+			return {
+				filters: {
+					docstatus: ['=', 1],
+					status: ['!=', 'Cancelled Booking'],
+					item_code: me.frm.doc.item_code,
+				}
+			 };
+		});
+	}
+
+	opportunity() {
+		this.frm.set_value("customer", null);
+		this.frm.set_value("lead", null);
+		frappe.call({
+			method: "erpnext.vehicles.doctype.vehicle_gate_pass.vehicle_gate_pass.get_opportunity_details",
+			args: {
+				"opportunity": this.frm.doc.opportunity,
+			},
+			callback: function (r) {
+				if (r.message && !r.exc) {
+					me.frm.set_value(r.message);
+				}
+			}
+		})
+	}
+
+	lead() {
+		frappe.call({
+			method: "erpnext.crm.doctype.lead.lead.get_lead_details",
```

Comment:
Use vehicle transaction controller's get customer details method .. do not use this method as it's out of date